runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningRockType.java

```diff
@@ -0,0 +1,172 @@
+import static net.runelite.api.ObjectID.ROCKS_7461;
+import static net.runelite.api.ObjectID.ROCKS_7484;
+import static net.runelite.api.ObjectID.ROCKS_7485;
+import static net.runelite.api.ObjectID.ROCKS_7486;
+import static net.runelite.api.ObjectID.ROCKS_7488;
+import static net.runelite.api.ObjectID.ROCKS_7489;
+import static net.runelite.api.ObjectID.ROCKS_7490;
+import static net.runelite.api.ObjectID.ROCKS_7491;
+import static net.runelite.api.ObjectID.ROCKS_7492;
+import static net.runelite.api.ObjectID.ROCKS_7493;
+import static net.runelite.api.ObjectID.ROCKS_7494;
+import static net.runelite.api.ObjectID.DEPLETED_VEIN_26665;
+import static net.runelite.api.ObjectID.DEPLETED_VEIN_26666;
+import static net.runelite.api.ObjectID.DEPLETED_VEIN_26667;
+import static net.runelite.api.ObjectID.DEPLETED_VEIN_26668;
+
+/**
+ * All the possible rock types the user can mine.
+ */
+public enum MiningRockType
+{
+
+	COPPER(0, "Copper", 1, 2.5, -1, false, true, ROCKS_7453, ROCKS_7484),
+	TIN(1, "Tin", 1, 2.5, -1, false, true, ROCKS_7485, ROCKS_7486),
+	IRON(2, "Iron", 15, 5.5, -1, false, true, ROCKS_7455, ROCKS_7488),
+	SILVER(3, "Silver", 15, 60, -1, false, true, ROCKS_7457, ROCKS_7490),
+	COAL(4, "Coal", 30, 29.5, -1, false, true, ROCKS_7456, ROCKS_7489),
+	GOLD(5, "Gold", 40, 59.5, -1, false, true, ROCKS_7458, ROCKS_7491),
+	MITHRIL(6, "Mithril", 55, 119.5, -1, false, true, ROCKS_7459, ROCKS_7492),
+	ADAMANTITE(7, "Adamantite", 70, 239.5, -1, false, true, ROCKS_7460, ROCKS_7493),
+	RUNITE(8, "Runite", 85, 720, -1, false, true, ROCKS_7461, ROCKS_7494),
+	AMETHYST(9, "Amethyst", 92, 75, -1, true, false, EMPTY_WALL),
+	PAY_DIRT(10, "Pay-Dirt", 30, 95, 125, true, false, DEPLETED_VEIN_26665, DEPLETED_VEIN_26666, DEPLETED_VEIN_26667, DEPLETED_VEIN_26668);
+
+	@Getter(AccessLevel.PACKAGE)
+	private final int index;
+
+	@Getter(AccessLevel.PACKAGE)
+	private final String name;
+
+	@Getter(AccessLevel.PACKAGE)
+	private final int requiredLevel;
+
+	@Getter(AccessLevel.PACKAGE)
+	private final boolean memberOnly;
+
+	@Getter(AccessLevel.PACKAGE)
+	private final boolean ground;
+
+	private final List<Integer> rocks;
+
+	@Getter(AccessLevel.PACKAGE)
+	private final double respawnTime, maxRespawnTime;
+
+	/**
+	 * @param index					Index, used to programmatically access element from an array.
+	 * @param name					Name to check for in chat & to display to player
+	 * @param level					The mining level required to mine this rock
+	 * @param respawnTime			The minimum respawn time for this rock
+	 * @param maxRespawnTime		The maximum respawn for this rock (-1 if no range)
+	 * @param memberOnly			If the rock can only be mined by members
+	 * @param ground				If the rock spawns on the ground or on a wall (true for ground, false for wall)
+	 * @param rocks					The RS IDs for the rocks. (Wall rocks uses the blank ID)
+	 */
+	MiningRockType(int index, String name, int level, double respawnTime, double maxRespawnTime, boolean memberOnly, boolean ground, int... rocks)
```

Comment:
Unless it's important that `rocks` be a `List`, I'd strongly prefer using `@AllArgsConstructor` on the enum rather than this ctor. Also, the arguments should use the same names and order as the defined fields.